validation/errors.go

```diff
@@ -0,0 +1,181 @@
+package validation
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// Error is an implementation of the 'error' interface, which represents a
+// field-level validation error.
+type Error struct {
+	Type     ErrorType
+	Field    string
+	BadValue interface{}
+	Detail   string
+}
+
+var _ error = &Error{}
```

Comment:
This variable was used for debug purpose and not used constantly. To avoid compilation error "not used variable"  you can use the blank identifier (_)
For example
```
package main

import (
    "fmt" // imported and not used: "fmt"
)

func main() {
    i := 1 // i declared and not used
}
```
```
package main

import (
    _ "fmt" // no more error
)

func main() {
    i := 1 // no more error
    _ = i
}
```



Because it is not used anymore I will remove it.